### ISSUE TITLE
`data` should not be required

### ISSUE THREAD

**Issue by dustinfarris** [JarrodCTaylor/fauxjax#8]
This is a feature request.

Right now, the user must supply a `data` key in order to match a request.  It would be nice if you could omit the key and have fauxjax match only what is provided.  (url, method, etc..)

**Comment by JarrodCTaylor** [comment]
Having it be strict was actually a goal of mine when I wrote this. I wanted to be overly vigilant about making sure my tests were doing what I intended for them to do and force myself to be explicit about all parts of the request. 

I am not opposed to possibly introducing this behavior as long as the functionality is opt in. Any ideas on how that api would look?

**Comment by dustinfarris** [comment]
I suppose a setting would be appropriate for opt-in.  Like:

```js
$.fauxjax.settings.lazyMatching = true;
```

**Comment by JarrodCTaylor** [comment]
I have a full week lined up, but I will give this some thought and take a closer look at it this weekend.

**Comment by JarrodCTaylor** [comment]
Closed with 96076a755862dff537c609e3737508da85d27585

In addition to the new API which differentiates between `request` and `response` when creating a new fauxjax handler. I have added a setting called `strictMatching` be default it is set to `true` if you set it to `false` you should have the functionality you describe above. Below is the expert from the updated API docs. Let me know if you run into any strangeness. 

## Strict Matching
By default fauxjax has `strictMatching` set to `true`. This means that fauxjax will expect the url, method, data and headers in a fauxjax instance to match all of those attributes on a real request in order to fake the request. You can tell fauxjax to be more lenient with the matching by setting 

``` javascript
$.fauxjax.settings.strictMatching = false
```

With `strictMatching` set to false fauxjax will only match the attributes that you provide to the `request` object in `$.fauxjax.new`. `url` will be the only required value.

**Comment by dustinfarris** [comment]
This is all really great.  I am thrilled with how this project is evolving.
